src/test/run-pass/command-pre-exec.rs

```diff
@@ -44,20 +43,6 @@ fn main() {
     let output = unsafe {
         Command::new(&me)
             .arg("test2")
-            .pre_exec(|| {
-                env::set_var("FOO", "BAR");
```

Comment:
I removed this test, since it was testing something one should not be doing in the first place, and with the new lock impl this caused a deadlock.

If the parent process is multi threaded, a child process is not supposed to call anything but syscalls after the `fork`. Here `env::set_var` will try to lock the `ENV_LOCK` mutex which triggers a deadlock because of how `parking_lot` is not fork-safe. See the comment around the `fork` call.

Reply:
This seems like a problem... I argued for `pre_exec` becoming unsafe for a reason, but it is still useful for extra "setup" that might have to be done before the `exec` in the forked-off process, and it doesn't seem too far-fetched to assume that someone might want to set env vars there. I know `Command` can do that, but someone might either not be aware or it doesn't really fit the way their code is written.